src/EnergyPlus/CurveManager.hh

```diff
@@ -306,26 +330,81 @@ namespace Curve {
     );
 
     void GetCurveMinMaxValues(EnergyPlusData &state,
-                              int CurveIndex,                          // index of curve in curve array
-                              Real64 &Var1Min,                         // Minimum values of 1st independent variable
-                              Real64 &Var1Max,                         // Maximum values of 1st independent variable
-                              ObjexxFCL::Optional<Real64> Var2Min = _, // Minimum values of 2nd independent variable
-                              ObjexxFCL::Optional<Real64> Var2Max = _, // Maximum values of 2nd independent variable
-                              ObjexxFCL::Optional<Real64> Var3Min = _, // Minimum values of 3rd independent variable
-                              ObjexxFCL::Optional<Real64> Var3Max = _, // Maximum values of 3rd independent variable
-                              ObjexxFCL::Optional<Real64> Var4Min = _, // Minimum values of 4th independent variable
-                              ObjexxFCL::Optional<Real64> Var4Max = _, // Maximum values of 4th independent variable
-                              ObjexxFCL::Optional<Real64> Var5Min = _, // Minimum values of 5th independent variable
-                              ObjexxFCL::Optional<Real64> Var5Max = _, // Maximum values of 5th independent variable
-                              ObjexxFCL::Optional<Real64> Var6Min = _, // Minimum values of 6th independent variable
-                              ObjexxFCL::Optional<Real64> Var6Max = _  // Maximum values of 6th independent variable
```

Comment:
Refactor this function with an overloaded set of functions. I didn't find any usages of this that didn't operate in terms of pairs, so there's probably a better way here, but I'll leave this for the larger reworking of CurveManager. I would suggest that it would make sense for this to be a part of the base class functionality (similar to the `value` methods) however it is changed.

Reply:
Yeah, I get it.  This curve refactor is going to look a little bloated and gross during the intermediate period, but it will lead to a much better situation.